src/parser/__init__.py

```diff
@@ -7,180 +8,215 @@
-                        case_soup = BeautifulSoup(file_handle, "html.parser", from_encoding="UTF-8")
-                    except Exception as e:
-                        print(f'error building beautiful soup for file {case_html_file_path}, {e}')
-                        raise e
-
-                # Get the county-specific parser class and method
-                parser_instance, parser_function = self.get_class_and_method(county=county)
-                if parser_instance is not None and parser_function is not None:
-                    case_data = parser_function(county, case_number, case_soup)
-                else:
-                    # Handle the case where parser_instance or parser_function is None
-                    print("Error: Could not obtain parser instance or function.")
-
-                # Adds county field to data
-                case_data['county'] = county
-
-                # Adds a hash to the JSON file of the underlying HTML
-                body = case_soup.find("body")
-                balance_table = body.find_all("table")[-1]
-                if "Balance Due" in balance_table.text:
-                    balance_table.decompose()
-                case_data["html_hash"] = xxhash.xxh64(str(body)).hexdigest()
-
-                self.write_json_data(case_json_path, case_number, case_data, test)
-
-            except Exception:
-                print(traceback.format_exc())
-                self.write_error_log(county, case_number)
-
-        RUN_TIME_PARSER = time() - START_TIME_PARSER
-        print(f"Parsing took {RUN_TIME_PARSER} seconds")
+        try:
+            # get input and output directories and make json dir if not present
+            case_html_path, case_json_path = self.get_directories(county, logger, test)
+
+            # start 
+            START_TIME_PARSER = time()
+            logger.info(f'Time started: {START_TIME_PARSER}')
+            # creating a list of json files already parsed 
+            cached_case_json_list = [
+                file_name.split(".")[0] for file_name in os.listdir(case_json_path)
+            ]
+
+            # Get a list of the HTML files that it needs to parse.
+            case_html_list = self.get_list_of_html(case_html_path, case_number, county, logger, parse_single_file)
+            logger.info(f'Starting for loop to parse {len(case_html_list)} cases')
+            for case_html_file_path in case_html_list:
+                try:
+                    case_number = os.path.basename(case_html_file_path).split(".")[0]
+
+                    logger.info(f"{case_number} - parsing")
+
+                    with open(case_html_file_path, 'r', encoding='utf-8', errors='ignore') as file:
+                        case_soup = BeautifulSoup(file, "html.parser")
+
+                    parser_instance, parser_function = self.get_class_and_method(county=county, logger=logger, test=test)   
+
+                    if parser_instance is not None and parser_function is not None:
```

Comment:
I think this is probably a carry over from the scraper structure, but Matt had a great point on my code there that ultimately I think that parser_instance and parser_function should be not None by the time it gets here and it probably isn't necessary to check this. Ultimately, this will error out anyway if you try to run None with the fields you're passing it.